### PR TITLE
[IMP] point_of_sale: add tax info on receipt

### PR DESCRIPTION
**Pull request by vlst-odoo**
Currently, the pos receipt shows the total amount of taxes paid. Different localisations require more information regarding the taxes to be shown on the receipt.

In this PR we improve the receipt by adding specific information for each of the applied taxes and a mechanism that would allow each localization to more easily display the proper tax information.

Task: 3305403

---

New tax breakdown as a result of this PR:

image

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/point_of_sale/static/src/app/store/models.js
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/store/models.js:776]: should remove this diff

File: addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.xml
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.xml:237]: I dont understand why there is this line ?

**Review by adgu-odoo — CHANGES_REQUESTED** [review]

**Review by adgu-odoo — APPROVED** [review]
Just need to remove the useless diff and squash the commits and that will be ok for me

**Comment by adgu-odoo** [comment]
@robodoo r+

**Comment by adgu-odoo** [comment]
After talking and testing with @caburj , we think it is good to remove the last line of the table tax table (the summary line) and keep the Total Taxes text under the Total on the receipt. After this change, you can keep the localization and we are sure the legislation for ae is still respected.

**Comment by vlst-odoo** [comment]
> After talking and testing with @caburj , we think it is good to remove the last line of the table tax table (the summary line) and keep the Total Taxes text under the Total on the receipt. After this change, you can keep the localization and we are sure the legislation for ae is still respected.

I agree, using the old `total` line would work, but what would be the advantage? I mean, why not merge what we have now, which respects the `ae` requirements, allows us to remove a whole module and also gives the breakdown of `total_without_tax` and `total_tax`.  This way, we also respect the same format as the receipt in the task. The upgrade script was already approved

**Comment by caburj** [comment]
> > After talking and testing with @caburj , we think it is good to remove the last line of the table tax table (the summary line) and keep the Total Taxes text under the Total on the receipt. After this change, you can keep the localization and we are sure the legislation for ae is still respected.
> 
> I agree, using the old `total` line would work, but what would be the advantage? I mean, why not merge what we have now, which respects the `ae` requirements, allows us to remove a whole module and also gives the breakdown of `total_without_tax` and `total_tax`. This way, we also respect the same format as the receipt in the task. The upgrade script was already approved

I guess the best action is to ask someone who know AE localization if this version is really compliant. I'll post a question in discord.

File: addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.js
Comment (caburj) [inline addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.js:60]: I think it's safer to assume that tax-letter mapping will be dependent on the whole tax object, not only on its tax amount. Redo this such that `getTaxLetter()` takes the tax object. Be default, it will just return an empty string. And other addons will override `getTaxLetter` to return the correct letter based on the tax record (object).

With this, we can get rid of the `getTaxLetterMapping` function.

(Am I inconsistent? Apology in advance.)

File: addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.js
Comment (vlst-odoo) [inline addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.js:52]: remove line

**Comment by caburj** [comment]
Kindly pass the tax object instead of the tax id. I noticed that in "getTaxLetter", the param description says the taxId is a number, but it's actually a string.

File: addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.xml
Comment (caburj) [reply addons/point_of_sale/static/src/app/screens/receipt_screen/receipt/receipt.xml:237]: It's okay to list only the first, according to dama. However, we'll merge this after merging the online payment. It just received the security override.

**Review by caburj — APPROVED** [review]
robodoo r+

**Comment by caburj** [comment]
robodoo r+

**Pull request closed without merging**
